e2e_tests/tests/fixtures/no_op/model_def.py

```diff
@@ -46,6 +46,7 @@ class NoOpTrialController(det.TrialController):
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
+        self.metric_writer = self.create_metric_writer()
```

Comment:
you probably need to change the TrialControllers that still define `create_metric_writer()` to not be @classmethods.  I think that includes NoOpTrialController, MetricsMakerTrialController, EstimatorTrialController, and TfKerasTrialController.